File: smarthome_libraries/DataLibrary/datalibrary/src/main/java/smarthome/library/datalibrary/store/listeners/DevicesObserver.java
package smarthome.library.datalibrary.store.listeners;

import smarthome.library.common.IotDevice;

import java.util.List;

public interface DevicesObserver {
    /**
     * @param devices devices whose data has been updated
     * @param isInnerCall indicates that data changed on subscriber (if true)
     */
    void onDevicesChanged(List<IotDevice> devices, boolean isInnerCall);
}

File: smarthome_libraries/DataLibrary/datalibrary/src/main/java/smarthome/library/datalibrary/store/listeners/DevicesObserver.kt
package smarthome.library.datalibrary.store.listeners

import smarthome.library.common.IotDevice

interface DevicesObserver {
    /**
     * @param devices devices whose data has been updated
     * @param isInnerCall indicates that data changed on subscriber (if true)
     */
    fun onDevicesChanged(devices: List<IotDevice>, isInnerCall: Boolean)
}